include/maliput_sparse/builder/builder.h

```diff
@@ -0,0 +1,360 @@
+/// of that process. The graph construction is a synchronous operation though and requires
+/// that the geometry information is available when building maliput::api::Lanes.
+///
+/// A super simple 2-lane dragway could be constructed as follows:
+///
+/// @code{cpp}
+/// const maliput::math:Vector3 start_left_lane_1{0., 0., 0.};
+/// const maliput::math:Vector3 end_left_lane_1{100., 0., 0.};
+/// const maliput::math:Vector3 start_right_lane_1{0., 5., 0.};
+/// const maliput::math:Vector3 end_right_lane_1{100., 5., 0.};
+/// const maliput::math:Vector3 start_left_lane_2 = start_right_lane_1;
+/// const maliput::math:Vector3 end_left_lane_2 = end_right_lane_1;
+/// const maliput::math:Vector3 start_right_lane_2{0., 10., 0.};
+/// const maliput::math:Vector3 end_right_lane_2{100., 10., 0.};
+/// const maliput::api::HBounds hbounds_l1{0., 5.};
+/// const maliput::api::HBounds hbounds_l2{0., 10.};
+///
+/// LineString3d left_line_string_1{start_left_lane_1, end_left_lane_1};
+/// LineString3d right_line_string_1{start_right_lane_1, end_right_lane_1};
+/// LineString3d left_line_string_2{start_left_lane_2, end_left_lane_2};
+/// LineString3d right_line_string_2{start_right_lane_2, end_right_lane_2};
+///
+/// RoadGeometryBuilder rg_builder;
+/// std::unique_ptr<RoadGeometry> road_geometry = RoadGeometryBuilder()
+///   .Id("two_lane_dragway")
+///   .StartJunction()
+///     .Id("j0")
+///     .StartSegment()
+///       .Id("j0_s0")
+///       .StartLane()
+///         .Id("j0_s0_l1")
+///         .HeightBounds(hbounds_l1)
+///         .StartLaneGeometry()
+///           .LeftLineString(left_line_string_1)
+///           .RightLineString(right_line_string_1)
+///         .EndLaneGeometry()
+///       .EndLane()
+///       .StartLane()
+///         .Id("j0_s0_l2")
+///         .HeightBounds(hbounds_l2)
+///         .StartLaneGeometry()
+///           .LeftLineString(left_line_string_2)
+///           .RightLineString(right_line_string_2)
+///         .EndLaneGeometry()
+///       .EndLane()
+///     .EndSegment()
+///   .EndJunction()
+///   .Build();
+/// @endcode{cpp}
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <maliput/api/junction.h>
+#include <maliput/api/lane.h>
+#include <maliput/api/road_geometry.h>
+#include <maliput/api/segment.h>
+#include <maliput/common/passkey.h>
+#include <maliput/geometry_base/junction.h>
+#include <maliput/geometry_base/lane.h>
+#include <maliput/geometry_base/road_geometry.h>
+#include <maliput/geometry_base/segment.h>
+
+#include "maliput_sparse/geometry/lane_geometry.h"
+#include "maliput_sparse/geometry/line_string.h"
+
+namespace maliput_sparse {
+namespace builder {
+namespace details {
+
+/// @brief Base class of the builder's hierarchy.
+/// @details Its sole purpose is to offer a template
+/// hierachy to avoid duplicating code that is related to storing the parent, returning
+/// it and making sure each children can call Passkey-protected methods.
```

Comment:
Sorry, I don't see what is BuilderBase class doing here. I think it is no longer needed. Could you enlighten me?

Reply:
Nothing now, good catch!